bridges/snowbridge/primitives/router/src/inbound/mod.rs

```diff
@@ -308,6 +338,36 @@ where
 			[GlobalConsensus(network), AccountKey20 { network: None, key: token.into() }],
 		)
 	}
+
+	fn convert_transact(
+		fee_asset_id: VersionedLocation,
+		message_id: H256,
+		chain_id: u64,
+		sender: H160,
+		origin_kind: OriginKind,
+		fee: u128,
+		weight_at_most: Weight,
+		payload: Vec<u8>,
+	) -> Result<(Xcm<()>, Balance), ConvertMessageError> {
+		let fee_asset_id: Location =
+			fee_asset_id.try_into().map_err(|_| ConvertMessageError::UnsupportedFeeAsset)?;
+		let xcm_fee: Asset = (fee_asset_id, fee).into();
+
+		let message = vec![
+			DescendOrigin(PalletInstance(InboundQueuePalletInstance::get()).into()),
+			// Change origin to the bridge.
+			UniversalOrigin(GlobalConsensus(Ethereum { chain_id })),
+			// DescendOrigin to the sender.
+			DescendOrigin(AccountKey20 { network: None, key: sender.into() }.into()),
+			WithdrawAsset(xcm_fee.clone().into()),
+			// Pay for execution.
+			BuyExecution { fees: xcm_fee, weight_limit: Unlimited },
+			Transact { origin_kind, require_weight_at_most: weight_at_most, call: payload.into() },
```

Comment:
We need to `RefundSurplus` execution left over from the Transact and `DepositAsset` any left over fees back into the prefunded account.

Reply:
Good catch! 
https://github.com/Snowfork/polkadot-sdk/pull/114/commits/ee6bdc5acd4dc8ad508d4e8ac115c4fafaad633c